guide/src/migration.md

```diff
@@ -15,6 +15,47 @@ For projects embedding Python in Rust, PyO3 no longer automatically initalizes a
 
 The limitation of the new default implementation is that it cannot support multiple `#[pymethods]` blocks for the same `#[pyclass]`. If you need this functionality, you must enable the `multiple-pymethods` feature which will switch `#[pymethods]` to the inventory-based implementation.
 
+### Deprecated `#[pyproto]` methods
+
+Some methods such as `__bytes__` and `__format__` have been possible to implement two ways in PyO3 for some time: via a `#[pyproto]` (e.g. `PyBasicProtocol` for the methods listed here), or by writing them directly in `#[pymethods]`. This is only true for a handful of the `#[pyproto]` methods (for technical reasons to do with the Python C-API).
+
+So as to have only one way to do things, and because the `#[pymethods]` macro has better documentation, error messages, and ergonomics, the `#[pyproto]` forms of these methods have been deprecated.
```

Comment:
The second reason seems redundant for me. If we have two similar codes, often the one grows better than the other.
```suggestion
So as to have only one way to do things, the `#[pyproto]` forms of these methods have been deprecated.
```